### DISCUSSION TITLE
New firmware release (v1.11.0) available on GitHub

### DISCUSSION THREAD

**Discussion by dmcnaugh** [thehighnibble/thehighnibble.github.io#38] [Announcements]
# [March 2023 Release](https://github.com/thehighnibble/firmware/releases/tag/v1.11.0)

This is a **minor** release that:

- Adds support for the [S-132 Add-on](https://thehighnibble.com/s132/)
- Adds **132 column mode** to `TTY:` devices (includes `TTY2:` & `TTY3:` for Cromemco Z-1)
- Adds **Tektronix 4014 vector display** (`TEK:`) as a slave to the `TTY:` VT100 terminal (only, not to `TTY2:` & `TTY3:` for Cromemco Z-1) 
- Fixes a stack overflow in the IMSAI & Cromemco firmware when `NVS_LOG_LEVEL` set to `INFO`
- Improves functionality in the `LIB:`, `MAN:` & `TAPE:` windows of the **Desktop UI**
- Fixes cut-and-paste into the `TTY:` devices (includes `TTY2:` & `TTY3:` for Cromemco Z-1)

See the release notes for more details.

**Comment by udo-munk** [comment]
Cool, Tek terminal now usable on z80pack machines.

**Comment by ElectricDream** [comment]
Hi Dave, 

I downloaded the new zip files from the release section, but they  seem to be broken .

I got "invalid archive" message when unzipping, under Linux and Windows.

**Reply by dmcnaugh** [reply]
My mistake, sorry about that. New files for FIRMWARE.zip and IMAGE.zip have been uploaded.

**Reply by ElectricDream** [reply]
Thanks ! Upgrade done :)

**Comment by GrantMeStrength** [comment]
Bug report:
When I open the TEK: view on Safari, I cannot resize the window. Fair enough - you stress that Safari is not supported.
When I open the TEK: view on Chrome, there is a white bar at the top of the window. This is on a Mac.
<img width="691" alt="Screenshot 2023-03-20 at 9 26 17 PM" src="https://user-images.githubusercontent.com/1928855/226517144-2ee3ba42-7f82-4493-91e7-b26b46536eec.png">

**Reply by GrantMeStrength** [reply]
Actually, the sizing seems off on other windows too:

<img width="830" alt="Screenshot 2023-03-20 at 9 28 17 PM" src="https://user-images.githubusercontent.com/1928855/226517337-f84a40f8-36d2-42aa-9e53-43805e21d834.png">

**Reply by dmcnaugh** [reply]
All very unexpected and peculiar. I am using Chrome on macOS as my main development platform and don't have this problem.
So we have to go through a triage process to understand the difference between your environment and mine.
- what is your Chrome version?
- what is you macOS version?
- can you try with all Chrome extensions disabled (you can log in as the Chrome guest user to simplify this)
- BTW. This production version is just a version number bump on the version you used successfully a couple of weeks ago, so I'm really confused as to what has changed. Was this a problem when you were on the Beta of the desktop UI in the last few weeks?

**Reply by GrantMeStrength** [reply]
I tried a different Mac. 
On a Mac running 13.3 beta, Chrome 106 (arm64), there is no issue.
On a Mac running 13.2.1, Chrome 110 (x86), when I log in as Guest, no issue. When I log in with my usual ID, there's an issue.
So there must be a plug-in that is causing a problem (i.e. not your firmware, sorry!) If I nail it down, I'll update this post.

Found the issue: I had tweaked the default text size in Chrome. Setting it back to medium, and all is good.
<img width="703" alt="Screenshot 2023-03-21 at 5 15 42 PM" src="https://user-images.githubusercontent.com/1928855/226770019-19e1b5b4-5eba-42e0-bcbe-acd63cff7c50.png">

**Reply by dmcnaugh** [reply]
Good job, I can look into whether I can make an update to my code to cope with that setting being changed from the default.
And add it to the known issues in the release notes.

**Reply by dmcnaugh** [reply]
Found a simple solution to this: had the title bar set to a fixed height of 30px, but can change it to be relative to the font size with 1.88em.
I will include this in the next update and you'll be able to change the Chrome default font size and it will still look right.

**Comment by tniccum21** [comment]
Firmware installed, smooth as silk!  The process is so easy with your setup.

**Comment by mbenson1** [comment]
I installed the new firmware, but I had to make the following changes to the instructions:

1.  The current esptool release (4.5.1) packages the python scripts as standalone executable elf images, so I had to change the "espefuse.py" to "espefuse"
2.  The current esptool release (4.5.1) doesn't have an installer and doesn't include instructions to manually install the file, so I elected to run the tool without installing it.  So I changed the "espefuse" to "./espefuse".  
3.  The SD card enumerated itself as ttyUSB0 on my system (Ubuntu 20.04 LTS), so I replaced "tty.SLAB_USBtoUART" to "ttyUSB0".

But in the end, I ran the flash.sh script anyway.  I had to make the same changes to the flash.sh script as well.

**Comment by mbenson1** [comment]
What do we do with the IMAGE package?  I have the files, but can't find anywhere in the files or online on what to do with them.  I'm looking for the SD card image.  Do I just copy these files to the SD card?  If so, how do I need to partition the SD card?  What file system format do I use?  Do I set any flags like the BOOT flag?  Is there any documentation anywhere?

**Reply by GrantMeStrength** [reply]
Yes, the contents of IMAGE go into the root of a FAT32 formatted SD card. Preferably on a PC to avoid all the hidden Mac files (which aren't hidden in the IMSAI/Cromemco system - they appear in LIB, Manual, conf files etc).

**Reply by udo-munk** [reply]
You can delete the .DS_Store files on the SD card, these are index files created and used by OSX finder.
